packages/widgets/src/lib/components/InferenceWidget/widgets/ConversationalWidget/ConversationalWidget.svelte

```diff
@@ -115,34 +149,41 @@
 		}
 	}
 
-	function isValidOutput(arg: any): arg is Response {
-		return (
-			arg && Array.isArray(arg?.conversation?.generated_responses) && Array.isArray(arg?.conversation?.past_user_inputs)
-		);
-	}
-
-	function parseOutput(body: unknown): {
-		conversation: Conversation;
+	function parseOutput(
+		body: unknown,
+		chat: Message[]
+	): {
+		chat: Message[];
 		output: Output;
 	} {
-		if (isValidOutput(body)) {
-			const conversation = body.conversation;
-			const pastUserInputs = conversation.past_user_inputs;
-			const generatedResponses = conversation.generated_responses;
-			const output = pastUserInputs
-				.filter(
-					(x, i) =>
-						x !== null && x !== undefined && generatedResponses[i] !== null && generatedResponses[i] !== undefined
-				)
-				.map((x, i) => ({
-					input: x ?? "",
-					response: generatedResponses[i] ?? "",
-				}));
-			return { conversation, output };
+		if (Array.isArray(body) && body.length) {
+			const text = body[0]?.generated_text ?? "";
+
+			if (!text.length) {
+				throw new TypeError("Model did not generate a response.");
```

Comment:
What's the reason for using `TypeError`? (Instead of the plain `Error` for example)

Reply:
My bad, I copied the error from somewhere else in the code. Will fix 👍